refractr/validate.py

```diff
@@ -4,44 +4,40 @@
 import re
 import sys
 import requests
-
-from urllib.parse import ParseResult
+import warnings
 
 from refractr.utils import *
+from refractr.url import URL
 from leatherman.dictionary import head_body
+from leatherman.repr import __repr__
 from leatherman.dbg import dbg
 
-
-# makes visualizing as string easier to read
-ParseResult.__repr__ = lambda self: self.geturl()
-
-def replace(pr, **kwargs):
-    return ParseResult(
-        scheme=kwargs.get('scheme', pr.scheme),
-        netloc=kwargs.get('netloc', pr.netloc),
-        path=kwargs.get('path', pr.path),
-        params=kwargs.get('params', pr.params),
-        query=kwargs.get('query', pr.query),
-        fragment=kwargs.get('fragment', pr.fragment))
-
 class Hop:
     def __init__(self, src, dst=None, status=None, match=None, ex=None):
         self.src = src
         self.dst = dst
         self.status = status
-        self.match = match
+        self._match = match
         self.ex = ex
 
+    __repr__ = __repr__
```

Comment:
Where is this assignment value coming from?

Reply:
https://github.com/scottidler/leatherman/blob/master/leatherman/repr.py which is a real package on pypi